plugin-flex-ts-template-v2/src/utils/feature-loader/logger.ts

```diff
@@ -0,0 +1,28 @@
+import FlexHelperSingleton from '../flex-helper';
+import logger from '../logger';
+import Destination from '../logger/destination';
+
+const destinations: Destination[] = [];
+
+export const init = () => {
+  const worker = FlexHelperSingleton.getCurrentWorker();
+  if (worker) {
+    logger.addMetaData('workerSid', worker.sid);
+    logger.addMetaData('workerName', worker.name);
+    logger.addMetaData('source', 'twilio:flex-ui');
+  }
+  logger.debug('logger initialized!');
+};
+
+export const processHooks = () => {
+  destinations.forEach((dest) => logger.addDestination(dest));
+  logger.processBuffer();
+};
```

Comment:
I think these functions can be combined into a single `init` function, as the metadata is not relevant until the buffer is flushed. Then, the feature loader can run the `init` function as part of the `execute deferred hooks` section of code. This makes things more consistent with the other hooks.

Reply:
Done! We now call init once and dont have to call any additional methods like other deferred hooks!